cpp/llm_chat.cc

```diff
@@ -989,19 +998,42 @@ class LLMChatModule : public ModuleNode {
 };
 
 tvm::runtime::Module CreateChatModule(tvm::runtime::Module executable,
-                                      const tvm::runtime::String& tokenizer_path,
+                                      std::unique_ptr<Tokenizer> tokenizer,
                                       const tvm::runtime::String& param_path, DLDevice device) {
   ObjectPtr<LLMChatModule> n = make_object<LLMChatModule>();
-  n->Init(executable, tokenizer_path, param_path, device);
+  n->Init(executable, std::move(tokenizer), param_path, device);
   return Module(n);
 }
 
+tvm::runtime::Module CreateChatModule(tvm::runtime::Module executable,
+                                      const tvm::runtime::String& tokenizer_path,
+                                      const tvm::runtime::String& param_path, DLDevice device) {
+  return CreateChatModule(executable, Tokenizer::FromFile(tokenizer_path), param_path, device);
+}
+
+tvm::runtime::Module CreateChatModule(tvm::runtime::Module executable,
+                                      const tvm::runtime::String& vocab_path,
+                                      const tvm::runtime::String& merges_path,
+                                      const tvm::runtime::String& param_path, DLDevice device) {
+  return CreateChatModule(executable, Tokenizer::FromBPE(vocab_path, merges_path), param_path,
+                          device);
+}
+
 // register as a system function that can be queried
 TVM_REGISTER_GLOBAL("mlc.llm_chat_create")
     .set_body_typed([](tvm::runtime::Module executable, const tvm::runtime::String& tokenizer_path,
                        const tvm::runtime::String& param_path, int device_type, int device_id) {
       return CreateChatModule(executable, tokenizer_path, param_path,
                               DLDevice{static_cast<DLDeviceType>(device_type), device_id});
     });
+
+TVM_REGISTER_GLOBAL("mlc.llm_chat_create_bpe_tokenizer")
```

Comment:
can we assume the common pattern and just provide a path?

Reply:
done.